File: services/graph4j_service/src/process/src/main/java/process/algorithm/Algorithms.java
package process.algorithm;

import lombok.RequiredArgsConstructor;
import process.algorithm.coloring.ColoringGeneticAlgorithm;
import process.algorithm.coloring.DSatur;
import process.algorithm.coloring.TabuCol;
import process.algorithm.shortestpath.FloydWarshall;
import process.algorithm.shortestpath.BellmanFord;
import process.algorithm.shortestpath.Dijkstra;
import process.instance.Instance;
import process.pubsub.Publisher;

import java.util.List;

@RequiredArgsConstructor
public class Algorithms {

    private final Instance instance;
    private final List<Publisher> publishers;

    public Algorithm getAlgorithm(String algorithmName) throws Exception {
        return switch (algorithmName) {
            case "Dijkstra" -> new Dijkstra(instance, publishers);
            case "Bellman-Ford" -> new BellmanFord(instance, publishers);
            case "Floyd-Warshall" -> new FloydWarshall(instance, publishers);
            case "DSatur" -> new DSatur(instance, publishers);
            case "TabuCol" -> new TabuCol(instance, publishers);
            case "Coloring Genetic Algorithm" -> new ColoringGeneticAlgorithm(instance, publishers);
            default -> throw new Exception("Algorithm not implemented");
        };
    }
}

File: services/jgrapht_service/src/algorithm/algorithms.py
from .algorithm import Algorithm


class Algorithms:
    def __init__(self, socket_id, file_id):
        self.algorithm_types = {
            "shortest_path": [
                Algorithm(socket_id, file_id, "Dijkstra", "shortest_path"),
                Algorithm(socket_id, file_id, "Bellman-Ford", "shortest_path"),
                Algorithm(socket_id, file_id, "A*", "shortest_path")
            ],
            "coloring": [
                Algorithm(socket_id, file_id, "TabuCol", "coloring"),
                Algorithm(socket_id, file_id, "DSatur", "coloring"),
            ]
        }

    def get(self, algorithm_type):
        if algorithm_type in self.algorithm_types:
            return self.algorithm_types[algorithm_type]

        return []